test/polygon/paraswap/paraswap.test.ts

```diff
@@ -0,0 +1,172 @@
+import BigNumber from "bignumber.js";
+import axios from "axios";
+const { waffle, ethers } = hre;
+const { provider, deployContract } = waffle;
+describe("Paraswap", function() {
+  const connectorName = "paraswap-test";
+  let dsaWallet0: any;
+  let masterSigner: any;
+  let instaConnectorsV2: any;
+  let connector: any;
+  const wallets = provider.getWallets();
+  const [wallet0, wallet1] = wallets;
+  before(async () => {
+    await hre.network.provider.request({
+        method: "hardhat_reset",
+        params: [
+            {
+                forking: {
+                    // @ts-ignore
+                    jsonRpcUrl: hre.config.networks.hardhat.forking.url,
+                    
+                },
+            },
+        ],
+    });
+    masterSigner = await getMasterSigner();
+    instaConnectorsV2 = await ethers.getContractAt(
+      abis.core.connectorsV2,
+      addresses.core.connectorsV2
+    );
+    connector = await deployAndEnableConnector({
+      connectorName,
+      contractArtifact: ConnectV2ParaswapV5Polygon__factory,
+      signer: masterSigner,
+      connectors: instaConnectorsV2,
+    });
+    console.log("Connector address", connector.address);
+  });
+  it("Should have contracts deployed.", async function() {
+    expect(!!instaConnectorsV2.address).to.be.true;
+    expect(!!connector.address).to.be.true;
+    expect(!!masterSigner.address).to.be.true;
+  });
+  describe("DSA wallet setup", function() {
+    it("Should build DSA v2", async function() {
+      dsaWallet0 = await buildDSAv2(wallet0.address);
+      expect(!!dsaWallet0.address).to.be.true;
+    });
+    it("Deposit ETH into DSA wallet", async function() {
+      await wallet0.sendTransaction({
+        to: dsaWallet0.address,
+        value: ethers.utils.parseEther("10"),
+      });
+
+      expect(await ethers.provider.getBalance(dsaWallet0.address)).to.be.gte(
+        ethers.utils.parseEther("10")
+      );
+    });
+  });
+  describe("Main", function() {
+    it("should swap successfully", async function() {
+      async function getArg() {
+        const slippage = 1;
+        /* matic -> USDT */
+        const sellTokenAddress = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"; // matic,  decimals 18
+        const sellTokenDecimals = 18;
+        const buyTokenAddress = "0xc2132d05d31c914a87c6611c10748aeb04b58e8f"; // USDT, decimals 6
+        const buyTokenDecimals = 6;
+        const amount = 2;
+        const srcAmount = new BigNumber(amount)
+          .times(new BigNumber(10).pow(sellTokenDecimals))
+          .toFixed(0);
+        const fromAddress = dsaWallet0.address;
+        let url = `https://apiv5.paraswap.io/prices/`;
+        let params = {
+          srcToken: sellTokenAddress,
+          destToken: buyTokenAddress,
+          srcDecimals: sellTokenDecimals,
+          destDecimals: buyTokenDecimals,
+          amount: srcAmount,
+          side: "SELL",
+          network: 137,
+        };
```

Comment:
I think they have changed the params for v5. 
check it here - https://developers.paraswap.network/api/v4-to-v5#1.-prices-endpoint